src/lib/utils.ts

```diff
@@ -65,3 +65,20 @@ export const throttledForEach = async <T, U>(
 	// We return the results of the iterator so the caller can await them as necessary
 	return await Promise.all(promises);
 };
+
+export const withRetries = async <T>(
+	func: () => Promise<T>,
+	delayDuration = 2000,
+	retries = 3,
+): Promise<T> => {
+	try {
+		return await func();
+	} catch (err) {
+		if (retries <= 1) {
```

Comment:
I thought that the `<= 0` was on purpose, like run this fn and retry N times.
But it seems that you meant the total attempts.
No strong opinion though. LGTM anyway.

Reply:
Hmm I thought of it as total retries (thinking of the first request as a try), but reading it now I think it makes sense what you're saying. I initially used `times` as a variable and didn't notice the subtle difference.